### DISCUSSION TITLE
We could use clang format to keep consistent formatting over the project

### DISCUSSION THREAD

**Discussion by Fabio3rs** [ggml-org/llama.cpp#868] [Ideas]
We could use the clang-format to generate a .clang-format file with formatting configurations and apply it on the CI, that will keep all the C/C++ code of the project with a consistent looking.

Cang format has some predefined settings that can be customized.
https://clang.llvm.org/docs/ClangFormat.html

And the .clang-format file is normally used by the clangd LSP to run the format command.

**Comment by sw** [comment]
I don't think you'd want the CI to mess with the source code, but it may be a good idea to add an additional check to CI that prevents merging messy formatting.

This could be achieved in a non-obtrusive manner by using [lint-diffs](https://github.com/AtakamaLLC/lint-diffs), which only checks modified lines. I was actually trying to get this to work on llama.cpp but with [cpplint](https://github.com/cpplint/cpplint), but it turns out that one is actually quite opinionated (in a not so good way IMO). clang-tidy may be a better choice.
